src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+import { updateTodo, deleteTodo } from '../../api/todos';
+
+interface Props {
+  todo?: Partial<Todo> | null;
+  isProcessed: boolean;
+  onDelete?: () => void;
+  onUpdate: (newTitle: string) => void | Promise<void | Partial<Todo>>;
```

Comment:
The `onUpdate` prop should have the return type `Promise<void> | void` instead of `void | Promise<void | Partial<Todo>>`. Since the function is supposed to update the title, it doesn't need to return the updated todo.